cmd/posix.go

```diff
@@ -391,22 +392,33 @@ func (s *posix) getVolDir(volume string) (string, error) {
 func (s *posix) getDiskID() (string, error) {
 	s.RLock()
 	diskID := s.diskID
+	fileInfo := s.formatFileInfo
+	lastCheck := s.formatLastCheck
 	s.RUnlock()
 
+	if fileInfo != nil && diskID != "" && time.Now().Before(lastCheck.Add(time.Second)) {
+		return diskID, nil
+	}
+
+	s.Lock()
+	defer s.Unlock()
+	if !s.formatLastCheck.Equal(lastCheck) {
```

Comment:
This line is not super clear for me, can you explain why this is useful ?

Reply:
`s.Lock()` will block all other than the first that get the lock. We let the first one finish and all the other blocked ones will not try to update, but returns in this block since the time was updated.

Adding a comment.